### PR TITLE
test: add unit test for createJobs deleteJobs

### PR DESCRIPTION
**Pull request by googs1025**
issue from #476

### CODE REVIEW

File: pkg/controllers/jobset_controller_test.go [hunk 1]
```diff
@@ -1150,3 +1161,1070 @@ func makeJob(args *makeJobArgs) *testutils.JobWrapper {
 		PodAnnotations(annotations)
 	return jobWrapper
 }
+
+type fakeClient struct {
```
Comment (danielvegamyhre) [inline pkg/controllers/jobset_controller_test.go:1165]: Can we move the helper functions and option structs (including `makeJob` and `makeJobArgs`) to the bottom of the file, and have the actual tests at the top?

Comment (googs1025) [reply pkg/controllers/jobset_controller_test.go:1165]: ya, that makes sense

**Pull request closed without merging**